lodash.js

```diff
@@ -2568,17 +2568,21 @@
      * @param {boolean[]} orders The sort orders of `props`.
      * @returns {Array} Returns the new sorted array.
      */
-    function baseSortByOrder(collection, props, orders) {
+    function baseSortByOrder(collection, comparitors, orders) {
       var index = -1,
           length = collection.length,
           result = isLength(length) ? Array(length) : [];
 
+      comparitors = arrayMap(comparitors, function(comparitor) {
+        return getCallback(comparitor, null, 1);
+      });
+
       baseEach(collection, function(value) {
         var length = props.length,
             criteria = Array(length);
 
         while (length--) {
-          criteria[length] = value == null ? undefined : value[props[length]];
+          criteria[length] = comparitors[length](value);
         }
```

Comment:
The one thing that bothers me is there's no `this` binding. It wouldn't be possible. This makes it a bit awkward compared to `_.sortBy`.